core/Symbols.cc

```diff
@@ -700,29 +701,40 @@ vector<Symbol::FuzzySearchResult> Symbol::findMemberFuzzyMatchConstant(const Glo
                 i++;
             }
             for (const auto scope : candidateScopes) {
-                for (auto member : scope.data(gs)->membersStableOrderSlow(gs)) {
+                scopeBest.clear();
+                for (auto member : scope.data(gs)->members()) {
                     if (member.first.kind() == NameKind::CONSTANT &&
                         member.first.dataCnst(gs)->original.kind() == NameKind::UTF8 && member.second.exists()) {
                         auto thisDistance = Levenstein::distance(
                             currentName, member.first.dataCnst(gs)->original.dataUtf8(gs)->utf8, best.distance);
                         if (thisDistance <= best.distance) {
+                            if (thisDistance < best.distance) {
+                                scopeBest.clear();
+                            }
                             best.distance = thisDistance;
                             best.symbol = member.second;
                             best.name = member.first;
-                            result.emplace_back(best);
+                            scopeBest.emplace_back(best);
                         }
                     }
                 }
+                if (!scopeBest.empty()) {
+                    // NOTE: Iteration order over members is nondeterministic, so we use SymbolId to deterministically
+                    // order the recommendations from this scope.
+                    // We order in decreasing symbol ID order because `result` is later reversed and we want earlier
+                    // ID'd symbols to be recommended first.
+                    fast_sort(scopeBest, [&](auto lhs, auto rhs) -> bool { return lhs.symbol._id > rhs.symbol._id; });
+                    for (auto item : scopeBest) {
+                        result.emplace_back(item);
+                    }
```

Comment:
I don't know whether the compiler would generate the same code here as with `auto &item`, but I think the latter is less confusing to reason about.  (Note that `auto &` is used when iterating over `globalBest`, below, so it doesn't seem completely unreasonable to use `auto &` here.)  Would you please use `auto &item` instead?